examples/dgfip_c/ml_primitif/main.ml

```diff
@@ -0,0 +1,94 @@
+
+open Common
+
+let read_test filename =
+
+  let test = Read_test.read_test filename in
+
+  List.fold_left (fun (tgv, res_prim) s ->
+      match s with
+      | `EntreesPrimitif pl ->
+        let tgv =
+          List.fold_left (fun tgv (code, montant) ->
+              try TGV.set tgv code montant
+              with Not_found ->
+                Printf.eprintf "Variable %s inexistante, test invalide\n" code;
+                exit 2
+            ) tgv pl
+        in
+        tgv, res_prim
+      | `ResultatsPrimitif pl ->
+        let res_prim =
+          List.fold_left (fun resultats (code, montant) ->
+              StrMap.add code montant resultats
+            ) res_prim pl
+        in
+        tgv, res_prim
+      | _ ->
+        tgv, res_prim
+    ) (TGV.empty, StrMap.empty) test
+
+let check_result tgv err expected_tgv expected_err =
+  let result = ref 0 in
+  StrMap.iter (fun code montant ->
+      (* Consider missing variables as 0.0 (DGFiP does the same) *)
+      let montant' = try TGV.get_def tgv code 0.0 with Not_found -> 0.0 in
+      if montant <> montant' then
+        (result := 1;
+         Printf.eprintf "KO | %s attendu: %f - calculé: %f\n"
+           code montant montant')
+    ) expected_tgv;
+  !result
+
+let run_test test_file =
+  Printf.printf "Testing %s...\n%!" test_file;
+
+  let tgv, res_prim  = read_test test_file in
+
+  (* Normalement fait par règle 1 *)
```

Comment:
Should be fixed by #169?

Reply:
I will try to remove that once #169 is merged.